hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/local/LocalRegionCache.java

```diff
@@ -214,11 +217,8 @@ public void onMessage(final Message<Object> message) {
 
     public boolean remove(final Object key) {
         final Expirable value = cache.remove(key);
-        if (value != null) {
-            maybeNotifyTopic(key, null, value.getVersion());
-            return true;
-        }
-        return false;
+        maybeNotifyTopic(key, null, (value != null) ? value.getVersion() : null);
```

Comment:
Both collections and entities can be cached on remote nodes in LOCAL mode even when absent from the current node's cache.  So I think this potentially increases the number of invalidation messages but is required.